fdbserver/TLogServer.actor.cpp

```diff
@@ -1055,27 +1061,35 @@ ACTOR Future<Void> tLogPopCore( TLogData* self, Tag inputTag, Version to, Refere
 	return Void();
 }
 
+ACTOR Future<Void> processPopRequests(TLogData* self, Reference<LogData> logData) {
+	state const size_t batchSize = 1000;
+	wait(self->toBePoppedMutex.take());
```

Comment:
It feels like it'd be easier and simpler to just `state std::map<Tag, Version> toBePopped; toBePopped = std::move(self->toBePopped)`.  Then you don't need to worry about the flowLock at all.  If by the time that the tLogPopCore is played for a particular pop, there's another ignorePops going on, then the pop will just be queued into self->toBePopped like any other pop.

Reply:
Good thought. And actually looking closer, I notice that all of the `tLogPopCore` calls here are passing in the exact tag and version from the map, which means that they would have been a no-op in the synchronous version of `tLogPopCore`. The only place I see that actually triggers this logic is at the bottom of `tLogPop`, so possibly we could move the `queuePopRequests` logic there and take the early wait out of `tLogPopCore` altogether.

Reply:
I like this approach better, I made the proposed change.